client/src/components/graph/graph.tsx

```diff
@@ -918,6 +918,7 @@ class Graph extends React.Component<{}, GraphState> {
           top: 0,
           left: 0,
         }}
+        data-test-id={`graph-wrapper-distance=${camera?.distance()}`}
```

Comment:
I needed to added the camera's distance into the test attribute. Otherwise, I had no way of figuring out the zoom level since it's a virtual zoom on the coordinates, not a viewport zoom.

Reply:
isn't this going to cause unnecessary re-renders?

Reply:
I figured if the camera is changing, the graph component would be re-rendered anyway. Let me double check if this is causing extra renders.

Reply:
Doesn't seem to be causing extra renders when zooming.

Reply:
After discussion with @tihuan, because the graph wrapper `div` element is re-rendering every time anything changes in the component anyway, including point position due to zoom, then adding dynamic `data-test-id` isn't causing extra renders. This is also confirmed by adding log statements to the component.

Reply:
Thanks for bringing this up, @seve! The `style={{...}}` is already re-generating a new POJO every time and causing re-rendering, so `data-test-id` changes is not making React render it more. It's already the most re-rendering React could do on this `<div>` 😄